tests/units/adapters/test_data_saver.py

```diff
@@ -45,3 +52,30 @@ def test_txt_writer_writes_file(tmpdir, mocker, students_data: list[Student], tx
     assert expected_file_path.exists()
     content = expected_file_path.read_text()
     assert 'Romek' in content
+
+
+def test_html_writer_writes_file(tmpdir, mocker, students_data: list[Student], html_writer) -> None:
+    mocker.patch('adapters.data_saver.render_content', return_value='<p>Romek</p>')
+
+    html_writer.write(students_data)
+
+    time_str = datetime.datetime.now().strftime('%Y%m%d%H%M%S')
+    expected_file_path = tmpdir.join(f'scores_{time_str}.html')
+
+    assert expected_file_path.exists()
+    content = expected_file_path.read_text()
+    assert '<p>Romek</p>' in content
+
+
+def test_render_content_txt(students_data):
+    template_name = 'templates/student_report_template.txt'
+    expected_content = f'Student Report\nGenerated at: {datetime.datetime.now().strftime("%Y-%m-%d %H:%M")}\n'
```

Comment:
Please add a case for rendering html. Preferably, as `@pytest.mark.fixture` with template name, and expected string given directly (not calculated in the test function)